system/handlers/core/AssetDownload.cfc

```diff
@@ -178,54 +184,55 @@ component {
 		return ReReplace( arguments.assetTitle, "\.#arguments.extension#$", "" ) & "." & arguments.extension;
 	}
 
-	private void function _checkDownloadPermissions( event, rc, prc ) output=false {
+	private boolean function _checkDownloadPermissions( event, rc, prc ) output=false {
 		var assetId        = rc.assetId       ?: "";
 		var derivativeName = rc.derivativeId  ?: "";
 
 		if ( Len( Trim( derivativeName ) ) && assetManagerService.isDerivativePubliclyAccessible( derivativeName ) ) {
-			return;
+			return false;
 		}
 
 		var permissionSettings = assetManagerService.getAssetPermissioningSettings( assetId );
 
+		if ( !permissionSettings.restricted ) {
+			return false;
+		}
+
 		if ( !event.isAdminUser() ) {
-			if ( permissionSettings.restricted ) {
-				if ( Len( Trim( permissionSettings.conditionId ) ) ) {
-					var conditionIsTrue = rulesEngineWebRequestService.evaluateCondition( permissionSettings.conditionId );
-
-					if ( !conditionIsTrue ) {
-						if ( !isLoggedIn() || ( permissionSettings.fullLoginRequired && isAutoLoggedIn() ) ) {
-							event.accessDenied( reason="LOGIN_REQUIRED", postLoginUrl=( cgi.http_referer ?: "" ) );
-						} else {
-							event.accessDenied( reason="INSUFFICIENT_PRIVILEGES" );
-						}
+			if ( Len( Trim( permissionSettings.conditionId ) ) ) {
+				var conditionIsTrue = rulesEngineWebRequestService.evaluateCondition( permissionSettings.conditionId );
+
+				if ( !conditionIsTrue ) {
+					if ( !isLoggedIn() || ( permissionSettings.fullLoginRequired && isAutoLoggedIn() ) ) {
+						event.accessDenied( reason="LOGIN_REQUIRED", postLoginUrl=( cgi.http_referer ?: "" ) );
+					} else {
+						event.accessDenied( reason="INSUFFICIENT_PRIVILEGES" );
 					}
-					return;
 				}
-				var hasPerm = event.isAdminUser() && hasCmsPermission(
-					  permissionKey       = "assetmanager.assets.download"
-					, context             = "assetmanagerfolder"
-					, contextKeys         = permissionSettings.contextTree
-					, forceGrantByDefault = IsBoolean( permissionSettings.grantAcessToAllLoggedInUsers ) && permissionSettings.grantAcessToAllLoggedInUsers
-				);
-				if ( hasPerm ) { return; }
+				return true;
+			}
+			var hasPerm = event.isAdminUser() && hasCmsPermission(
```

Comment:
Not sure what this check is trying to achieve, but it will never be run - we've already established that the user is not an admin user...

Reply:
Yup, that was hungover from previous refactors I guess. Refactored now.